src/main/java/org/icgc_argo/workflow/relay/util/NextflowDocumentConverter.java

```diff
@@ -56,19 +58,30 @@ public static WorkflowDocument buildWorkflowDocument(
 
     val engineParams =
         EngineParameters.builder()
-            .revision(workflow.getRevision())
-            .resume(workflow.getResume())
-            .launchDir(workflow.getLaunchDir())
-            .projectDir(workflow.getProjectDir())
-            .workDir(workflow.getWorkDir())
+            .revision(ObjectUtils.isNotEmpty(workflow.getRevision())
+                ? workflow.getRevision()
+                : oldDocument.getEngineParameters().getRevision())
+            .resume(ObjectUtils.isNotEmpty(workflow.getResume())
+                ? workflow.getResume()
+                : oldDocument.getEngineParameters().getResume())
+            .launchDir(ObjectUtils.isNotEmpty(workflow.getLaunchDir())
+                ? workflow.getLaunchDir()
+                : oldDocument.getEngineParameters().getLaunchDir())
+            .projectDir(ObjectUtils.isNotEmpty(workflow.getProjectDir())
+                ? workflow.getProjectDir()
+                : oldDocument.getEngineParameters().getProjectDir())
+            .workDir(ObjectUtils.isNotEmpty(workflow.getWorkDir())
+                ? workflow.getWorkDir()
+                : oldDocument.getEngineParameters().getWorkDir())
```

Comment:
This uses a lot of repeated code that makes it hard to read the actual logic. Can we use a method like `defaultIfNull` instead? 
example:
```java
.revision(ObjectUtils.defaultIfNull(workflow.getRevision(), oldDocument.getEngineParameters().getRevision())
```

If we need to check for empty as well as null, can we make our own util method `defaultIfEmpty`

Reply:
https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/ObjectUtils.html#defaultIfNull-T-T-

Reply:
That makes so much sense. Made it concise. Hope this one is more readable.